0_locs_poly_setup/src/get_polygons.R

```diff
@@ -0,0 +1,47 @@
+#' Function to download NHDPlusHR HUC-4 geopackages that contain all NHD layers from 
+#' a HUC
+#' 
+#' @param HUC a 4-digit or longer HUC value
+#' @param minimum_sqkm a numeric threshold value to filter waterbodies, in square 
+#' kilometers; if specified as NA_real_, no filter will be applied
+#' @returns filepath for new {sf} file that contains the filtered polygons
+#' 
+#' 
+#' 
+get_polygons <-  function(HUC, minimum_sqkm) {
+  #check for out folder, create if not present
+  if (dir.exists('0_locs_poly_setup/out/') == FALSE) {
+    dir.create('0_locs_poly_setup/out/')
+  }
+  
+  # create temp directory for download of HR .gdb files
+  try(dir.create('tmp'))
```

Comment:
question: Is `try()` necessary here? I'm not sure if the same result would happen without this but it seems possible?

Reply:
question: Also, is `tmp` truly temporary? It seems to persist in my local version

Reply:
Great question -- I had a line `unlink('tmp', recursive = T)` (it was right after the return of this function) at the end of this function, but it would never remove the folder. It made me think that {targets} didn't have the ability to run this type of code where it eliminates something? Because it didn't delete the tmp folder at the end of this, you are absolutely right, it persists locally, and is therefore, definitely not temporary.

I'm going to try just running that line as a target to see if it will eliminate the folder. Otherwise I need to level up and use a legit temporary folder.

Reply:
Okay, calling it in its own target works properly!

Reply:
...but only if there is a singlular folder in the folder..... putting on my big boy data science pants and using tempdir().

Reply:
Oh no! Haha. I'm surprised by that behavior but it's helpful to know!

Reply:
And following back around on this: now I also know why you were asking why it was wrapped in a `try()` - to close the loop: I thought this error'ed and not warn'ed. :)

Reply:
AH!